equinox/nn/_attention.py

```diff
@@ -179,8 +211,44 @@ def __init__(
         - `dtype`: The dtype to use for all trainable parameters in this layer.
             Defaults to either `jax.numpy.float32` or `jax.numpy.float64` depending
             on whether JAX is in 64-bit mode.
+        - `kv_cache`: The callable module to handle KV caching.
         - `key`: A `jax.random.PRNGKey` used to provide randomness for parameter
             initialisation. (Keyword only argument.)
+
+
+
+        !!! info
+
+        The following shows an example of how to use `MultiheadAttention` for
+        autoregressive decoding with a `kv_cache` and `state`:
+        ```python
+        import equinox as eqx
+        import jax
+
+        query_size = 6
+        num_heads = 1
+        state_length = 8
+        seq_len = 3
+
+        standard_kv_cache = eqx.nn.StandardKVCache(
+            key_shape=(state_length, num_heads, query_size),
+            value_shape=(state_length, num_heads, query_size),
+        )
+
+        mha, state = eqx.nn.make_with_state(MultiheadAttention)(
+            query_size=query_size,
+            num_heads=num_heads,
+            kv_cache=standard_kv_cache,
+            key=jax.random.PRNGKey(0)
+
+        )
+
+        for states in range(state_length):
+            x = jax.numpy.ones(shape=(seq_len, query_size)) * (states + 1)
+            y, state = mha(x, x, x, mask="causal", state=state)
+
+        ```
+
```

Comment:
(Another another spurious newline here too.)

Think of all those precious bytes wasted on GitHub... !